src/sdk/PnP.Core/Model/SharePoint/Core/Internal/ListDataAsStreamHandler.cs

```diff
@@ -692,6 +693,10 @@ private static object GetJsonPropertyValue(JsonElement propertyValue, FieldType
                         {
                             return null;
                         }
+                        else if (propertyValue.ValueKind == JsonValueKind.Object)
+                        {
+                            return propertyValue.ToObject<System.Dynamic.ExpandoObject>();
```

Comment:
this is a "just in case" for other field types that happen to be treated as objects